pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test: TestLogic_create_as failed

pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test.TestLogic_create_as [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5892521?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5892521?buildTab=artifacts#/) on master @ [e52ca86863fcb7c56816f7c867b6663582700c81](https://github.com/cockroachdb/cockroach/commits/e52ca86863fcb7c56816f7c867b6663582700c81):


```
=== RUN   TestLogic_create_as
    test_log_scope.go:162: test logs captured to: /artifacts/tmp/_tmp/e469329b4a975cd977ed70727d62cbe5/logTestLogic_create_as2461467964
    test_log_scope.go:80: use -show-logs to present logs inline
[08:31:32] rng seed: -5192782918951519379

    logic.go:2838: 
        
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1969/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-dbg/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/create_as:392: 
        expected success, but found
        (XXUUU) command is too large: 7081637 bytes (max: 5000000)
        replica_raft.go:251: in evalAndPropose()
[08:31:54] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1969/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-dbg/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/create_as with config fakedist-disk: 77 tests, 1 failures
    logic.go:3570: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1969/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-dbg/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/create_as:392: error while processing
    logic.go:3570: pq: command is too large: 7081637 bytes (max: 5000000)
    panic.go:500: -- test log scope end --
--- FAIL: TestLogic_create_as (23.26s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_create_as.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-18101

pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test.TestLogic_create_as [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5918335?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5918335?buildTab=artifacts#/) on master @ [cd113bf428e8868c38b2f2f295210f5c30d13a2d](https://github.com/cockroachdb/cockroach/commits/cd113bf428e8868c38b2f2f295210f5c30d13a2d):


```
=== RUN   TestLogic_create_as
    test_log_scope.go:162: test logs captured to: /artifacts/tmp/_tmp/e469329b4a975cd977ed70727d62cbe5/logTestLogic_create_as2098425773
    test_log_scope.go:80: use -show-logs to present logs inline
[08:09:40] setting distsql_workmem='32419B';
[08:09:40] rng seed: -144865159823433323

[08:09:42] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2312/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-dbg/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/create_as: 75 statements
    logic.go:2838: 
        
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2312/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-dbg/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/create_as:392: 
        expected success, but found
        (XXUUU) command is too large: 7081043 bytes (max: 5000000)
        replica_raft.go:251: in evalAndPropose()
[08:10:18] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2312/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-dbg/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/create_as with config fakedist-disk: 77 tests, 1 failures
    logic.go:3570: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2312/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-dbg/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/create_as:392: error while processing
    logic.go:3570: pq: command is too large: 7081043 bytes (max: 5000000)
    panic.go:500: -- test log scope end --
--- FAIL: TestLogic_create_as (38.85s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_create_as.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/create_table.go
+++ b/pkg/sql/create_table.go
@@ -579,9 +579,10 @@ func (n *createTableNode) startExec(params runParams) error {
 					break
 				}
 
-				// Periodically flush out the batches, so that we don't issue gigantic,
+				// Periodically flush out the batches, so that we don't issue gigantic
 				// raft commands.
-				if ti.currentBatchSize >= ti.maxBatchSize {
+				if ti.currentBatchSize >= ti.maxBatchSize ||
+					ti.b.ApproximateMutationBytes() >= ti.maxBatchByteSize {
 					if err := tw.flushAndStartNewBatch(params.ctx); err != nil {
 						return err
 					}

